services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx

```diff
@@ -69,24 +121,100 @@ export const ReleaseDialog: React.FC<ReleaseDialogProps> = (props) => {
                         <span className={classNames('release-dialog-commitId', className)}>
                             {release.undeployVersion === undefined ? 'undeploy version' : release?.sourceCommitId}
                         </span>
-                        <Button onClick={setClosed} className={classNames('release-dialog-close', className)}>
-                            <svg
-                                width="20"
-                                height="20"
-                                viewBox="0 0 20 20"
-                                fill="none"
-                                xmlns="http://www.w3.org/2000/svg">
-                                <path d="M1 1L19 19M19 1L1 19" stroke="white" strokeWidth="2" strokeLinecap="round" />
-                            </svg>
-                        </Button>
+                        <Button
+                            onClick={setClosed}
+                            className={classNames('release-dialog-close', className)}
+                            icon={
+                                <svg
+                                    width="20"
+                                    height="20"
+                                    viewBox="0 0 20 20"
+                                    fill="none"
+                                    xmlns="http://www.w3.org/2000/svg">
+                                    <path
+                                        d="M1 1L19 19M19 1L1 19"
+                                        stroke="white"
+                                        strokeWidth="2"
+                                        strokeLinecap="round"
+                                    />
+                                </svg>
+                            }
+                        />
                     </div>
-                    <List className={classNames('release-env-list', className)}>
-                        {envs.map((env) => (
-                            <ListItem button key={env}>
-                                <ListItemText primary={env} />
-                            </ListItem>
+                    <ul className={classNames('release-env-list', className)}>
+                        {sortEnvironmentsByUpstream(envs).map((env) => (
```

Comment:
please extract some of this code into its own componet, e.g. `<EnvironmentList> or <Environment>`

Reply:
Created EnvironmentList component!